### DISCUSSION TITLE
filter_sensitive_data + binary POST?

### DISCUSSION THREAD

**Discussion by reidab** [vcr/vcr#871] [Ideas]
I have some tests that exercise an upload API, and I'd prefer that that blobs that I'm POST-ing weren't persisted in my VCR cassettes. It seemed like the `filter_sensitive_data` facility might work well for this, but my attempts so far haven't worked out.

I've been trying things along the lines of:

``` ruby
VCR.configure do |c|
  c.filter_sensitive_data('<VIDEO_BLOB>') do |interaction|
    interaction.request.uri.include?('upload') && interaction.request.body
    # interaction.request.uri.include?('upload') && Base64.encode64(interaction.request.body)
  end
end
```

Any ideas?

**Comment by mcfiredrill** [comment]
Hmm. I use this for filtering S3 keys. I haven't seen people try to use it to filter the entire body.

According to the docs:

```
The block should return the sensitive text that you want replaced with the substitution string. 
```

Are you sure the request body the exact same every time?
